azurerm/internal/services/dataprotection/data_protection_backup_instance_postgresql_resource.go

```diff
@@ -0,0 +1,224 @@
+				ResourceType:     utils.String("Microsoft.DBForPostgreSQL/servers"),
+				ResourceURI:      utils.String(""),
+			},
+			FriendlyName: utils.String(id.Name),
+			PolicyInfo: &dataprotection.PolicyInfo{
+				PolicyID: utils.String(policyId.ID()),
+			},
+		},
+	}
+	future, err := client.CreateOrUpdate(ctx, id.BackupVaultName, id.ResourceGroup, id.Name, parameters)
+	if err != nil {
+		return fmt.Errorf("creating/updating DataProtection BackupInstance (%q): %+v", id, err)
+	}
+
+	if err := future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("waiting for creation/update of the DataProtection BackupInstance (%q): %+v", id, err)
+	}
+
+	stateConf := &pluginsdk.StateChangeConf{
+		Pending:    []string{string(dataprotection.ConfiguringProtection), string(dataprotection.UpdatingProtection)},
+		Target:     []string{string(dataprotection.ProtectionConfigured)},
+		Refresh:    policyProtectionStateRefreshFunc(ctx, client, id),
+		MinTimeout: 1 * time.Minute,
+		Timeout:    d.Timeout(pluginsdk.TimeoutCreate),
+	}
+
+	if _, err = stateConf.WaitForStateContext(ctx); err != nil {
+		return fmt.Errorf("waiting for BackupInstance(%q) policy protection to be completed: %+v", id, err)
+	}
+
+	d.SetId(id.ID())
+	return resourceDataProtectionBackupInstancePostgreSQLRead(d, meta)
+}
+
+func resourceDataProtectionBackupInstancePostgreSQLRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).DataProtection.BackupInstanceClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.BackupInstanceID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, id.BackupVaultName, id.ResourceGroup, id.Name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] dataprotection %q does not exist - removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("retrieving DataProtection BackupInstance (%q): %+v", id, err)
+	}
+	d.Set("name", id.Name)
+	d.Set("resource_group_name", id.ResourceGroup)
+	d.Set("vault_name", id.BackupVaultName)
+	if props := resp.Properties; props != nil {
+		if props.DataSourceInfo != nil {
+			d.Set("database_id", props.DataSourceInfo.ResourceID)
+			d.Set("location", props.DataSourceInfo.ResourceLocation)
+		}
+		if props.PolicyInfo != nil {
+			d.Set("backup_policy_id", props.PolicyInfo.PolicyID)
+		}
+	}
+	return nil
+}
+
+func resourceDataProtectionBackupInstancePostgreSQLDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).DataProtection.BackupInstanceClient
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.BackupInstanceID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	future, err := client.Delete(ctx, id.BackupVaultName, id.ResourceGroup, id.Name)
+	if err != nil {
+		if response.WasNotFound(future.Response()) {
+			return nil
+		}
```

Comment:
this hides a bug - can we remove this?

Reply:
Yes, I'll remove it.

Reply:
yes, but that's not what I mean - this hides a bug in the Terraform Configuration where there's a missing dependency - this is something we'll be removing from all resources, as such we shouldn't add this to new resources

Reply:
ie if the resource was not found the delete failed so we should just error

Reply:
Thanks! I'll remove it,